src/software/geom/algorithms/rasterize.cpp

```diff
@@ -0,0 +1,245 @@
+#include "software/geom/algorithms/rasterize.h"
+
+#include <algorithm>
+#include <cmath>
+
+#include "software/geom/algorithms/contains.h"
+#include "software/geom/algorithms/distance.h"
+#include "software/geom/algorithms/intersection.h"
+#include "software/geom/algorithms/rasterize.h"
+
+std::vector<Point> rasterize(const Circle& circle, const double resolution_size)
+{
+    // Iterating through the x values from left (origin - radius) to right (origin +
+    // radius) of the circle which are resolution_size apart. Then finding the min and max
+    // y values for each x value using:
+    //             y = +-sqrt(r^2 - (x - k)^2) + h      (<== (y - h)^2 + (x - k)^2 = r^2)
+    // and filling in the points resolution_size away from each other between them.
+    std::vector<Point> covered_points;
+
+    // Offset for the radius to avoid points being outside of the circle due to floating
+    // point errors
+    const double EPSILON = 0.0001;
+    double radius        = circle.radius() - EPSILON;
+    double diameter      = radius * 2;
+    Point origin         = circle.origin();
+
+    // max number of pixels in each dimension
+    int max_num_pixels = (int)std::ceil(diameter / resolution_size);
```

Comment:
nit: we use static_cast<int> for compile time guarantee, please look for other instances of this

Reply:
updated to static_cast